pages/price-feeds/use-real-time-data/sui.mdx

```diff
@@ -1,82 +1,124 @@
+</Tabs>
 
-// In order to use Pyth prices in your protocol you need to submit the price update data to Pyth contract in your target
-// chain. `getPriceUpdateData` creates the update data which can be submitted to your contract.
+Pyth also provides a javascript SDK to construct transaction blocks that update price feeds:
 
-const priceUpdateData = await connection.getPriceFeedsUpdateData(priceIds);
-```
+<Tabs items={["npm", "yarn"]}>
+  <Tabs.Tab>
+    ```sh copy npm install --save @pythnetwork/pyth-sui-js ```
+  </Tabs.Tab>
+  <Tabs.Tab>```sh copy yarn add @pythnetwork/pyth-sui-js ```</Tabs.Tab>
+</Tabs>
 
-## On-chain prices
+## Write Contract Code
 
-### ⚠️ **_Important Note for Integrators_**
+<Callout type="warning" emoji="⚠️">
 
-Your Sui Move module **should NOT** have a hard-coded call to `pyth::update_single_price_feed`. In other words, the Sui Pyth `pyth::update_single_price_feed` entry point should never be called by a contract, instead it should be called directly from client code (e.g. Typescript or Rust).
+Your Sui Move module **should NOT** have a hard-coded call to `pyth::update_single_price_feed.` In other words, a contract should **never call** the Sui Pyth `pyth::update_single_price_feed` entry point. Instead, it should be called directly from client code (e.g., Typescript or Rust).
 
-This is because when a Sui contract is [upgraded](https://docs.sui.io/build/package-upgrades), the new address is different from the original. If your module has a hard-coded call to `pyth::update_single_price_feed` living at a fixed call-site, it may eventually get bricked due to the way Pyth upgrades are implemented. (We only allows users to interact with the most recent package version for security reasons).
+When Sui contracts are [upgraded](<(https://docs.sui.io/build/package-upgrades)>), the address changes, which makes the old address no longer valid. If your module has a hard-coded call to `pyth::update_single_price_feed` living at a fixed call-site, it may eventually get bricked due to how Pyth upgrades are implemented. (We only allow users to interact with the most recent package version for security reasons).
 
 Therefore, you should build a [Sui programmable transaction](https://docs.sui.io/build/prog-trans-ts-sdk) that first updates the price by calling `pyth::update_single_price_feed` at the latest call-site from the client-side and then call a function in your contract that invokes `pyth::get_price` on the `PriceInfoObject` to get the recently updated price.
-You can use `SuiPythClient` to build such transactions and handles all the complexity of updating the price feeds.
+You can use `SuiPythClient` to build such transactions and handle all the complexity of updating the price feeds.
+
+Consult [Fetch Price Updates](../fetch-price-updates) for more information on how to fetch the `pyth_price_update`.
+
+</Callout>
 
-### Update Example
+The code snippet below provides an example of how to update the Pyth price feeds:
```

Comment:
I think the organization of these code snippets is confusing. Most of our guides have the contract code first, then they don't have the client code at all. In this case, maybe the contract code first, then the client code as a separate section (sort of how we used to have it before?)

Reply:
I agree, but especially for sui, it is recommended not to update the price from the contract, I believe we keep this flow.